optuna/pruners/percentile.py

```diff
@@ -93,6 +96,9 @@ def prune(self, study, trial):
         if step is None:
             return False
 
+        if step % self.interval_steps != 0:
```

Comment:
IMO, it's better to take `n_warm_up_steps` into account, that is as follows:
```suggestion
        if (step + n_warmup_steps) % self.interval_steps != 0:
```

Reply:
Thanks for the suggestion, I applied you suggestion. Let me update the unit tests and clarify the behavior in the documentation.